xdsl/dialects/eqsat.py

```diff
@@ -70,10 +77,73 @@ def verify_(self) -> None:
                     "another eclass."
                 )
 
+    def delete_operand(self, index: int) -> None:
```

Comment:
This should definitely be on Rewriter, not an operation, let's move that there and add tests?